packages/bus-core/src/service-bus/service-bus.integration.ts

```diff
@@ -129,18 +134,32 @@ describe('ServiceBus', () => {
     })
 
     describe('when a message is successfully handled from the queue', () => {
-      it('should delete the message from the queue', async () => {
+      beforeAll(async () => {
+        messageReadMiddleware.reset()
+
+        messageReadMiddleware
+          .setup(x => x(It.isAny(),It.isAny()))
+          .returns((_, next) => next())
+          .verifiable(Times.once())
+
         callback.reset()
         callback
           .setup(c => c())
           .callback(() => undefined)
           .verifiable(Times.once())
         await sut.publish(event)
         await sleep(10)
+      })
```

Comment:
can this be done without `sleep`? I find it introduces test flakiness 

```suggestion
        callback.reset()
        
        await new Promise<void>(resolve => {
          callback
              .setup(c => c())
              .callback(resolve)
              .verifiable(Times.once())
            await sut.publish(event)
          })
        })
```